### PR TITLE
Add decimal version of round

### LINKED ISSUE #2605: TRUNCATE() result error
### Steps to reproduce the behavior (Required)
`SELECT TRUNCATE(0.0, 1326124691);`

### Expected behavior (Required)
0.000000000000000000000000000000
### Real behavior (Required)
```
+-------------------------+
| truncate(0, 1326124691) |
+-------------------------+
|                     nan |
+-------------------------+
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
select current_version();
+------------------------------------------+
| current_version()                        |
+------------------------------------------+
| PIPELINE_SQLANCER_MASTER_RELEASE 635482f |
+------------------------------------------+
```

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #2605

## Problem Summary(Required) ：

Related-pr 
* [3744](https://github.com/StarRocks/starrocks/pull/3744)
* [3993](https://github.com/StarRocks/starrocks/pull/3993)

* Add decimal version of round. 
* Always apply decimal version first.
* Downgrade to double version if second param is neither int literal nor slotRef expression.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/DecimalV3FunctionAnalyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|4|4|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -513,7 +513,13 @@ public Void visitFunctionCall(FunctionCallExpr node, Scope scope) {
                     throw new SemanticException("Time Type can not used in %s function",
                             fnName);
                 }
-            } else if (Arrays.stream(argumentTypes).anyMatch(Type::isDecimalV3)) {
+            } else if (FunctionSet.decimalRoundFunctions.contains(fnName) ||
+                    Arrays.stream(argumentTypes).anyMatch(Type::isDecimalV3)) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:517]: need the or condition?

File: be/src/exprs/vectorized/math_functions.cpp [hunk 2]
```diff
@@ -447,7 +447,8 @@ void MathFunctions::decimal_truncate(const int128_t& lv, const int32_t& l_scale,
     }
 }
 
-ColumnPtr MathFunctions::truncate_decimal128(FunctionContext* context, const starrocks::vectorized::Columns& columns) {
+template <DecimalRoundRule rule>
+ColumnPtr MathFunctions::decimal_round(FunctionContext* context, const Columns& columns) {
```
Comment (kangkaisen) [inline be/src/exprs/vectorized/math_functions.cpp:451]: Why not improve for const column？

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java (hunk 1 above)
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:517]: yes, comments right below

File: be/src/exprs/vectorized/math_functions.cpp (hunk 2 above)
Comment (liuyehcf) [reply be/src/exprs/vectorized/math_functions.cpp:451]: Emm, what do you mean? I didn't get it

Comment (kangkaisen) [reply be/src/exprs/vectorized/math_functions.cpp:451]: Ignore me. you have improved for const column

**Review by kangkaisen — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Review by Pslydhh — APPROVED** [review]

File: be/src/exprs/vectorized/math_functions.cpp [hunk 3]
```diff
@@ -428,17 +428,17 @@ void MathFunctions::decimal_truncate(const int128_t& lv, const int32_t& l_scale,
             // E.g. 1.2345 --(scale up by 2)--> 1.234500 --(scale down by 2) --> 1.2345
             *res = lv;
         } else {
-            (*is_over_flow) |= DecimalV3Cast::to_decimal_truncate<int128_t, int128_t, int128_t, true, true>(
```
Comment (stdpain) [inline be/src/exprs/vectorized/math_functions.cpp:431]: DecimalV3Cast::to_decimal_truncate may be unused any more?

**Review by satanson — APPROVED** [review]

**Pull request merged**